### DISCUSSION TITLE
Not able to connect to server .

### DISCUSSION THREAD

**Discussion by AnshTandon0** [php-amqplib/php-amqplib#1086] [Help wanted]
I have hosted a rabbitMQ server . When trying to connect to it through AMQPS . It is giving error . 

**<?php

require_once __DIR__ . '/vendor/autoload.php';
use PhpAmqpLib\Connection\AMQPSSLConnection;
use PhpAmqpLib\Message\AMQPMessage;

$ssl_options = array(
    'capath' => 'C:\Users\Ansh\Desktop\server_test' ,
    'cafile' => 'ca.crt', 
    'verify_peer' => true,
    'SNI_enabled' => false
);  

$connection = new AMQPSSLConnection('mq1.egrabber.net',5671,'job-q1','h5x80tfgcfdeawaj5SDggl-all','jobs-host-01' , $ssl_options );
$channel = $connection->channel();

$channel->queue_declare('rpc_queue', false, false, false, false);

function fib($n)
{
    if ($n == 0) {
        return 0;
    }
    if ($n == 1) {
        return 1;
    }
    return fib($n-1) + fib($n-2);
}

echo " [x] Awaiting RPC requests\n";
$callback = function ($req) {
    $n = intval($req->body);
    echo ' [.] fib(', $n, ")\n";

    $msg = new AMQPMessage(
        (string) fib($n),
        array('correlation_id' => $req->get('correlation_id'))
    );

    $req->delivery_info['channel']->basic_publish(
        $msg,
        '',
        $req->get('reply_to')
    );
    $req->ack();
};

$channel->basic_qos(null, 1, null);
$channel->basic_consume('rpc_queue', '', false, false, false, false, $callback);

while ($channel->is_open()) {
    $channel->wait();
}

$channel->close();
$connection->close();

?>**

PHP Fatal error:  Uncaught PhpAmqpLib\Exception\AMQPIOException: Error Connecting to server(0):   in C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php:114 Stack trace: #0 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AbstractConnection.php(253): PhpAmqpLib\Wire\IO\StreamIO->connect() #1 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AbstractConnection.php(235): PhpAmqpLib\Connection\AbstractConnection->connect() #2 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AMQPStreamConnection.php(64): PhpAmqpLib\Connection\AbstractConnection->__construct('job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', false, 'AMQPLAIN', Object(PhpAmqpLib\Wire\AMQPWriter), 'en_US', Object(PhpAmqpLib\Wire\IO\StreamIO), 0, 3, 0, NULL) #3 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AMQPSSLConnection.php(37): PhpAmqpLib\Connection\AMQPStreamConnection->__construct('mq1.egrabber.ne...', 5671, 'job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', false, 'AMQPLAIN', NULL, 'en_US', 3, 130, Resource id #18, false, 0, 0, 'ssl', NULL) #4 C:\Users\Ansh\Desktop\server_test\rpc_server.php(13): PhpAmqpLib\Connection\AMQPSSLConnection->__construct('mq1.egrabber.ne...', 5671, 'job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', Array) #5 {main}   thrown in C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php on line 114  Fatal error: Uncaught PhpAmqpLib\Exception\AMQPIOException: Error Connecting to server(0):   in C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php:114 Stack trace: #0 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AbstractConnection.php(253): PhpAmqpLib\Wire\IO\StreamIO->connect() #1 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AbstractConnection.php(235): PhpAmqpLib\Connection\AbstractConnection->connect() #2 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AMQPStreamConnection.php(64): PhpAmqpLib\Connection\AbstractConnection->__construct('job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', false, 'AMQPLAIN', Object(PhpAmqpLib\Wire\AMQPWriter), 'en_US', Object(PhpAmqpLib\Wire\IO\StreamIO), 0, 3, 0, NULL) #3 C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AMQPSSLConnection.php(37): PhpAmqpLib\Connection\AMQPStreamConnection->__construct('mq1.egrabber.ne...', 5671, 'job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', false, 'AMQPLAIN', NULL, 'en_US', 3, 130, Resource id #18, false, 0, 0, 'ssl', NULL) #4 C:\Users\Ansh\Desktop\server_test\rpc_server.php(13): PhpAmqpLib\Connection\AMQPSSLConnection->__construct('mq1.egrabber.ne...', 5671, 'job-q1', 'h5x80tfgcfdeawa...', 'jobs-host-01', Array) #5 {main}   thrown in C:\Users\Ansh\Desktop\server_test\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php on line 114

**Comment by danijel-resc** [comment]
Post some code, otherwise, no one can help you. This error means that the socket connection could not be opened. Or, in more detail, the `stream_socket_client` returned `false`.

**Comment by AnshTandon0** [comment]
@danijel-resc I have added the code . Can you please check now and suggest something ?
